framework/include/kernels/GenericKernel.h

```diff
@@ -27,3 +27,25 @@ class GenericKernel<true> : public ADKernel
   static InputParameters validParams() { return ADKernel::validParams(); };
   GenericKernel(const InputParameters & parameters) : ADKernel(parameters) {}
 };
+
+#define usingGenericKernelMembers                                                                  \
+  usingFunctionInterfaceMembers;                                                                   \
+  usingPostprocessorInterfaceMembers;                                                              \
+  usingMooseObjectMembers;                                                                         \
+  usingTransientInterfaceMembers;                                                                  \
+  usingTaggingInterfaceMembers;                                                                    \
+  usingBlockRestrictableMembers;                                                                   \
+  using GenericKernel<is_ad>::_qp;                                                                 \
+  using GenericKernel<is_ad>::_i;                                                                  \
+  using GenericKernel<is_ad>::_j;                                                                  \
+  using GenericKernel<is_ad>::_u;                                                                  \
+  using GenericKernel<is_ad>::_phi;                                                                \
+  using GenericKernel<is_ad>::_test;                                                               \
+  using GenericKernel<is_ad>::_q_point;                                                            \
+  using GenericKernel<is_ad>::_var;                                                                \
+  using GenericKernel<is_ad>::_name;                                                               \
+  using GenericKernel<is_ad>::getVar;                                                              \
+  using Coupleable::coupled;                                                                       \
+  using Coupleable::coupledComponents;                                                             \
+  using Coupleable::coupledGenericValue;                                                           \
+  using MaterialPropertyInterface::getGenericMaterialProperty
```

Comment:
You can also remove both of these because they are templates whose template argument can't be deduced by the method arguments (e.g. you will always need to do `this->template getGenericMaterialProperty<true/false>`)